### PR TITLE
Initial ERC 1155 random tokenId implementation

### PR DESCRIPTION
**Pull request by kneelsdev**

### CODE REVIEW

File: contracts/ArchetypeLogic.sol [hunk 1]
```diff
@@ -430,4 +408,49 @@ library ArchetypeLogic {
...
+    uint256[] memory tokenSupply,
+    uint32[] memory maxSupply,
+    uint32[] memory validIds,
+    uint256 quantity
+  ) public view returns (uint256[] memory) {
+    if (validIds.length == 0) {
+      validIds = new uint32[](maxSupply.length);
+      for (uint256 i = 0; i < validIds.length; i++) {
+        validIds[i] = uint32(i) + 1;
+      }
```
Comment (coindegen) [inline contracts/ArchetypeLogic.sol:422]: Isn't this pricy to compute on each mint?

Comment (kneelsdev) [reply contracts/ArchetypeLogic.sol:422]: Need to do some more testing, really depends on how many tokenIds are in the collection. Will be expensive for a large collection.

Comment (kneelsdev) [reply contracts/ArchetypeLogic.sol:424]: But it is all in memory, so may not be too bad.

Comment (kneelsdev) [reply contracts/ArchetypeLogic.sol]: see updated implementation @coindegen

File: test/archetype.test.ts [hunk 2]
```diff
@@ -2064,6 +2064,58 @@ describe("Factory", function () {
 
     await expect(await nftMint.totalSupply()).to.be.equal(2);
   });
+
+  it("test erc115 test batch mint verification", async function () {
+    const [accountZero, accountOne, accountTwo] = await ethers.getSigners();
+    let default_config = { ...DEFAULT_CONFIG };
+    default_config.maxSupply = new Array(500).fill(10); // 500 tokenIds, 10 each
+    default_config.maxBatchSize = 1000;
```
Comment (coindegen) [inline test/archetype.test.ts:2072]: eslint flags this 
image

Use 
```
    const default_config = {
      ...DEFAULT_CONFIG,
      maxSupply: new Array(500).fill(10),
      maxBatchSize: 1000,
    };
 ```

File: test/archetype.test.ts [hunk 3]
```diff
@@ -1679,18 +1731,161 @@ describe("Factory", function () {
...
       tokenAddress: ZERO,
     });
 
     await nftMint.connect(minter2).mint({ key: HASH256, proof: [] }, 40, ZERO, "0x", { value: 0 });
 
-    await expect(await nftMint.totalSupply()).to.be.equal(DEFAULT_CONFIG.maxSupply);
+    await expect(await nftMint.totalSupply()).to.be.equal(100);
+  });
+
+  it("test erc115 specific tokenId mints", async function () {
```
Comment (coindegen) [inline test/archetype.test.ts:1746]: fix typo. should be erc1155 not erc115.
actually tripped me up because I was searching for the 1155 tests :)

**Pull request closed without merging**